docs/sec-blueprint/08-Hardening.md

```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
+other advantage of this approach is that multiple applications can share
+the same resources at the same time.
+
+Root access **should not be allowed** for the following utilities:
 
-```bash
-  CONFIG_SECURITY_SELINUX_DEVELOP=n
-  CONFIG_SECURITY_SELINUX_DISABLE=n
-  CONFIG_SECURITY_SELINUX_BOOTPARAM=n
+```
+  login
+  su
+  ssh
+  scp
+  sftp
 ```
 
+Root access **should not be allowed** for the console device. The
+development environment should allow users to login with pre-created
+user accounts.
+
+Switching to elevated privileges shall be allowed in the development
+environment via sudo.
+
+Network Hardening
+-----------------
+
+### Disable IPv4 Forwarding
+
+  The net.ipv4.ip\_forward sysctl setting controls if IP forwarding is
+  allowed or not on the System. Unless the system is used as a router or
+  gateway, IPv4 forwarding should be disabled.
+
+### Disable IP Source Routing
+
+  Disable IP source routing on all interfaces through the
+  net.ipv4.conf.\*.accept\_source\_route = 0 setting.
+ 
+  IP source routing would allow a remote user (the sender) to specify
+  the route that the packet should take, rather than use the (default)
+  routing tables used by the routers between the sender and the
+  destination. This could be used to spoof IP addresses and still get
+  the replies (rather than sending the replies to the real owner of the
+  IP address).
+
+### Disable ICMP 
+
+  Use of ICMP, especially ping, shall be avoided. This is a common way
+  of gaining access to a system.
+
+#### Disable ICMP Redirects
+
+  Set net.ipv4.conf.\*.accept\_redirects=0 to disable ICMP redirect
+  support on the interfaces.
+ 
+  ICMP redirect messages are used by routers to inform hosts to use a
+  different gateway than the one used. These packets should only be sent
+  by the gateway of the system. In managed and embedded devices the
+  gateway is controlled and any changes should be controlled.
+ 
+  Allowing ICMP redirect messages would allow for "remote" updating of
+  the routing table, which could allow an attacker to get all packets
+  sent to the outside first rather than the packets immediately going to
+  the real gateway.
+
+#### Ignore ICMP Echo Broadcasts
+
+  When net.ipv4.icmp\_echo\_ignore\_broadcasts=1 is set, then your
+  system will not reply to broadcast “ping” requests.
+
+#### Ignore ICMP Bogus Error Responses
+
+  When an invalid response is given to broadcast frames (which occurs
+  sometimes in erroneous routers), the Linux kernel will by default log
+  this event. These can be disabled by setting
+  throughnet.ipv4.icmp\_ignore\_bogus\_error\_responses to 1.
+
+### Ignore all broadcast message
+
+  All the IP packets that come on the address “255.255.255.255” shall be
+  ignored. This can be done through the iptables rules.
+
+### Disable IPV6 
+
+  If there are no plans of using IPV6, it is a good practice to disable
+  this support as it will reduce the size of the kernel TCP/IP stack.
```

Comment:
humm... isn't it time to switch to ipv6 and disable ipv4? (joke) I fully agree with the assertion.